packages/next-swc/crates/core/src/hook_optimizer.rs

```diff
@@ -74,7 +74,7 @@ impl HookOptimizer {
         } = &decl;
         let init_clone = init.clone();
         if let Pat::Array(a) = name {
-            if let Expr::Call(c) = &*init.as_deref().unwrap() {
+            if let Expr::Call(c) = init.as_deref().unwrap() {
```

Comment:
@kdy1 the changes in this commit are following suggestions from clippy. Looks like toolchain update brings few more rules.